crates/router/src/connector/riskified/transformers/api.rs

```diff
@@ -0,0 +1,601 @@
+    requires_shipping: Option<bool>,
+    product_id: Option<String>,
+    category: Option<String>,
+    brand: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Serialize, Eq, PartialEq, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum Source {
+    DesktopWeb,
+}
+
+#[derive(Debug, Deserialize, Serialize, Eq, PartialEq, Clone)]
+pub struct RiskifiedMetadata {
+    vendor_name: String,
+    shipping_lines: Vec<ShippingLines>,
+}
+
+impl TryFrom<&frm_types::FrmCheckoutRouterData> for RiskifiedPaymentsCheckoutRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(payment_data: &frm_types::FrmCheckoutRouterData) -> Result<Self, Self::Error> {
+        let metadata: RiskifiedMetadata = payment_data
+            .frm_metadata
+            .clone()
+            .ok_or(errors::ConnectorError::MissingRequiredField {
+                field_name: "frm_metadata",
+            })?
+            .parse_value("Riskified Metadata")
+            .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+
+        let billing_address = payment_data.get_billing_address_with_phone_number()?;
+        let shipping_address = payment_data.get_shipping_address_with_phone_number()?;
+        let address = payment_data.get_billing_address()?;
+
+        Ok(Self {
+            order: CheckoutRequest {
+                id: payment_data.attempt_id.clone(),
+                email: payment_data.request.email.clone(),
+                created_at: common_utils::date_time::now(),
+                updated_at: common_utils::date_time::now(),
+                gateway: payment_data.request.gateway.clone(),
+                total_price: payment_data.request.amount,
+                cart_token: payment_data.attempt_id.clone(),
+                line_items: payment_data
+                    .request
+                    .get_order_details()?
+                    .iter()
+                    .map(|order_detail| LineItem {
+                        price: order_detail.amount,
+                        quantity: i32::from(order_detail.quantity),
+                        title: order_detail.product_name.clone(),
+                        product_type: order_detail.product_type.clone(),
+                        requires_shipping: order_detail.requires_shipping,
+                        product_id: order_detail.product_id.clone(),
+                        category: order_detail.category.clone(),
+                        brand: order_detail.brand.clone(),
+                    })
+                    .collect::<Vec<_>>(),
+                source: Source::DesktopWeb,
+                billing_address: OrderAddress::try_from(billing_address).ok(),
+                shipping_address: OrderAddress::try_from(shipping_address).ok(),
+                total_discounts: 0,
+                currency: payment_data.request.currency,
+                referring_site: "hyperswitch.io".to_owned(),
+                discount_codes: [DiscountCodes {
```

Comment:
total_discounts is zero and still we are passing `discount_codes ` is this valid?

Reply:
Must be removed. Removing it.